homeassistant/components/freedompro/light.py

```diff
@@ -0,0 +1,147 @@
+"""Support for Freedompro light."""
+import json
+import math
+
+from pyfreedompro import put_state
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    LightEntity,
+)
+from homeassistant.const import CONF_API_KEY
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import COORDINATOR, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro light."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
+    devices = [
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "lightbulb"
+    ]
+
+    async_add_entities(devices, False)
+
+
+class Device(CoordinatorEntity, LightEntity):
+    """Representation of an Freedompro light."""
+
+    def __init__(self, hass, api_key, device, coordinator):
+        """Initialize the Freedompro light."""
+        super().__init__(coordinator)
```

Comment:
Save the session so that you do not have to request it every time for `put_state`.
`self._session = aiohttp_client.async_get_clientsession(self._hass)`.